File: backend/src/main/java/ma/investpro/repository/FournisseurRepository.java
package ma.investpro.repository;

import ma.investpro.entity.Fournisseur;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface FournisseurRepository extends JpaRepository<Fournisseur, Long> {

    Optional<Fournisseur> findByCode(String code);

    Optional<Fournisseur> findByIce(String ice);

    List<Fournisseur> findByActifTrue();

    List<Fournisseur> findByNonResident(Boolean nonResident);

    boolean existsByCode(String code);

    boolean existsByIce(String ice);

    @Query("SELECT f FROM Fournisseur f WHERE " +
           "LOWER(f.code) LIKE LOWER(CONCAT('%', :search, '%')) OR " +
           "LOWER(f.raisonSociale) LIKE LOWER(CONCAT('%', :search, '%')) OR " +
           "LOWER(f.ice) LIKE LOWER(CONCAT('%', :search, '%'))")
    List<Fournisseur> search(@Param("search") String search);
}

File: backend/src/main/kotlin/ma/investpro/repository/FournisseurRepository.kt
package ma.investpro.repository

import ma.investpro.entity.Fournisseur
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import org.springframework.stereotype.Repository

@Repository
interface FournisseurRepository : JpaRepository<Fournisseur, Long> {

    /**
     * Recherche par code unique
     */
    fun findByCode(code: String): Fournisseur?

    /**
     * Recherche par ICE (Identifiant Commun de l'Entreprise)
     */
    fun findByIce(ice: String): Fournisseur?

    /**
     * Recherche par raison sociale (contient)
     */
    fun findByRaisonSocialeContainingIgnoreCase(raisonSociale: String): List<Fournisseur>

    /**
     * Recherche les fournisseurs actifs
     */
    fun findByActifTrue(): List<Fournisseur>

    /**
     * Recherche les fournisseurs non-résidents
     */
    fun findByNonResidentTrue(): List<Fournisseur>

    /**
     * Recherche les fournisseurs résidents
     */
    fun findByNonResidentFalse(): List<Fournisseur>

    /**
     * Vérifier si un code existe
     */
    fun existsByCode(code: String): Boolean

    /**
     * Vérifier si un ICE existe
     */
    fun existsByIce(ice: String): Boolean

    /**
     * Recherche multi-critères
     */
    @Query("""
        SELECT f FROM Fournisseur f
        WHERE (:raisonSociale IS NULL OR f.raisonSociale ILIKE CONCAT('%', :raisonSociale, '%'))
        AND (:nonResident IS NULL OR f.nonResident = :nonResident)
        AND (:actif IS NULL OR f.actif = :actif)
        ORDER BY f.raisonSociale ASC
    """)
    fun search(
        @Param("raisonSociale") raisonSociale: String?,
        @Param("nonResident") nonResident: Boolean?,
        @Param("actif") actif: Boolean?
    ): List<Fournisseur>
}
